test/CMakeLists.txt

```
@@ -49,7 +49,13 @@ function(celeritas_add_device_test base)
   if(CELERITAS_USE_CUDA OR CELERITAS_USE_HIP)
     set(_cuda_args GPU SOURCES "${base}.test.cu")
   endif()
-  celeritas_add_test("${base}.test.cc" ${_cuda_args} ${ARGN})
+    # GCC 13 requires some tests to have -ldl explicitely added
+    if(${base} MATCHES "Range"
+      OR ${base} MATCHES "StackAllocator"
+      OR ${base} MATCHES "ObserverPtr")
+          set(ARGN LINK_LIBRARIES dl)
```

If it is needed, it should be spelt `${CMAKE_DL_LIBS}`
```suggestion
          set(ARGN LINK_LIBRARIES ${CMAKE_DL_LIBS})
```

---

Echo: https://github.com/celeritas-project/celeritas/pull/1587#discussion_r1921267188